internal/remoteconfig/remoteconfig.go

```diff
@@ -91,9 +91,12 @@ type Client struct {
 	stop       chan struct{}
 
 	callbacks             []Callback
+	_callbacksMu          sync.RWMutex
 	products              map[string]struct{}
+	_productsMu           sync.RWMutex
 	productsWithCallbacks map[string]ProductCallback
 	capabilities          map[Capability]struct{}
+	_capabilitiesMu       sync.RWMutex
```

Comment:
nit: is there any reason for prefixing these fields with underscore?

Reply:
This is a personal habit, I can remove the prefixes

Reply:
It's fine. Let me review it again.

Reply:
Hm, I think this sets a bad precedence. Using underscore prefixes for go variable names, especially on structs, is highly unorthodox. It'd be nice if we could not do this going forward.